spec/osut_tests_spec.rb

```diff
@@ -1333,1134 +1129,2447 @@ module M
+    attic_floor = attic_floor.get
+
+    expect(core.nameString.include?("Core")).to be(true)
+    # 22.69, 13.46, 0,                        !- X,Y,Z Vertex 1 {m}
+    # 22.69,  5.00, 0,                        !- X,Y,Z Vertex 2 {m}
+    #  5.00,  5.00, 0,                        !- X,Y,Z Vertex 3 {m}
+    #  5.00, 13.46, 0;                        !- X,Y,Z Vertex 4 {m}
+    # -----,------,--
+    # 17.69 x 8.46 = 149.66 m2
+    expect(core.floorArea).to be_within(TOL).of(149.66)
+    core_volume = core.floorArea * core.ceilingHeight
+    expect(core_volume).to be_within(TOL).of(core.volume)
+    expect(attic.volume).to be_within(TOL).of(720.19)
+    expect(attic.floorArea).to be_within(TOL).of(567.98)
+
+    expect(mod1.poly(core_floor, true).empty?).to be(false)   # convex
+    expect(mod1.poly(core_ceiling, true).empty?).to be(false) # convex
+    expect(mod1.poly(attic_floor, true).empty?).to be(false)  # convex
+    expect(mod1.status.zero?).to be(true)
+
+    # Insert new 'mini' (2m x 2m) floor/ceiling at the centre of the existing
+    # core space. Initial insertion resorting strictly to adding leader lines
+    # from the initial core floor/ceiling vertices to the new 'mini'
+    # floor/ceiling.
+    centre = OpenStudio::getCentroid(core_floor.vertices)
+    expect(centre.empty?).to be(false)
+    centre = centre.get
+    mini_w = centre.x - 1 # 12.845
+    mini_e = centre.x + 1 # 14.845
+    mini_n = centre.y + 1 # 10.230
+    mini_s = centre.y - 1 #  8.230
+    mini_floor_vtx = OpenStudio::Point3dVector.new
+    mini_floor_vtx << OpenStudio::Point3d.new(mini_e, mini_n, 0)
+    mini_floor_vtx << OpenStudio::Point3d.new(mini_e, mini_s, 0)
+    mini_floor_vtx << OpenStudio::Point3d.new(mini_w, mini_s, 0)
+    mini_floor_vtx << OpenStudio::Point3d.new(mini_w, mini_n, 0)
+    mini_floor = OpenStudio::Model::Surface.new(mini_floor_vtx, model)
+    mini_floor.setName("Mini floor")
+    expect(mini_floor.outsideBoundaryCondition).to eq("Ground")
+    expect(mini_floor.setSpace(core)).to be(true)
+
+    mini_ceiling_vtx = OpenStudio::Point3dVector.new
+    mini_ceiling_vtx << OpenStudio::Point3d.new(mini_w, mini_n, 3.05)
+    mini_ceiling_vtx << OpenStudio::Point3d.new(mini_w, mini_s, 3.05)
+    mini_ceiling_vtx << OpenStudio::Point3d.new(mini_e, mini_s, 3.05)
+    mini_ceiling_vtx << OpenStudio::Point3d.new(mini_e, mini_n, 3.05)
+    mini_ceiling = OpenStudio::Model::Surface.new(mini_ceiling_vtx, model)
+    mini_ceiling.setName("Mini ceiling")
+    expect(mini_ceiling.setSpace(core)).to be(true)
+
+    mini_attic_vtx = OpenStudio::Point3dVector.new
+    mini_attic_vtx << OpenStudio::Point3d.new(mini_e, mini_n, 3.05)
+    mini_attic_vtx << OpenStudio::Point3d.new(mini_e, mini_s, 3.05)
+    mini_attic_vtx << OpenStudio::Point3d.new(mini_w, mini_s, 3.05)
+    mini_attic_vtx << OpenStudio::Point3d.new(mini_w, mini_n, 3.05)
+    mini_attic = OpenStudio::Model::Surface.new(mini_attic_vtx, model)
+    mini_attic.setName("Mini attic")
```

Comment:
<img width="1293" alt="xtrafloor" src="https://github.com/rd2/osut/assets/98104985/e8c3ef08-4518-4d95-bff2-12e5295476b0">

The _attic core_ floor (as well as the _core ceiling_) vertices are not yet modified at this stage.

Reply:
Yes, SketchUp will detect this as an inner loop of the larger polygon, there is code that then runs to turn inner loops into subsurfaces (auto-classified as door, window, or skylight).  Is that what you are seeing when you open this in SketchUp?  It will be turned into a skylight?  The way around that would be to add vertices to the outer polygon (like you did here https://github.com/rd2/osut/pull/13#discussion_r1276124965) so the smaller surface is technically adjacent and not "inside" the larger one.

Reply:
No, SketchUp has difficulty with the vertex sequencing I illustrate in [#13 (comment)](https://github.com/rd2/osut/pull/13#discussion_r1276124965). SketchUp (2021) refuses to illustrate the OSM geometry altogether (I believe a single 3D line ends up floating in 3D space, that's it).